### PR TITLE
fix(Core/Player): Fix Player not starting at full health

### LINKED ISSUE #15115: Death Knights do not start with full health.
### Current Behaviour

Death knights are created with close to 70% of their health pool, immediately having to regenerate health.

### Expected Blizzlike Behaviour

Death Knights should be created with full health.

### Source

[Wrath Retail](https://www.youtube.com/watch?v=ADiaJS585RM?t=53).

### Steps to reproduce the problem

1. Create a Death Knight.
2. Observe.

### Extra Notes

_No response_

### AC rev. hash/commit

84dc9e2ac8182e53dfca1540b3eb4f1a601128ea

### Operating system

Windows 10 x64

### Custom changes or Modules

[mod-ptr-template](https://github.com/heyitsbench/mod-ptr-template)
[mod-worgoblin](https://github.com/heyitsbench/mod-worgoblin)

### PR DESCRIPTION
**Pull request by sudlud**
## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes #15115

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -686,6 +686,10 @@ bool Player::Create(ObjectGuid::LowType guidlow, CharacterCreateInfo* createInfo
     }
     // all item positions resolved
 
+    // ensure player starts with full health
+    UpdateAllStats();
+    SetFullHealth();
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/Player.cpp:691]: https://github.com/azerothcore/azerothcore-wotlk/blob/508b1658fcbc4e625f939da75963480be2354430/src/server/game/Entities/Player/Player.cpp#L589-L591

We already do this in Player::Create

Comment (sudlud) [reply src/server/game/Entities/Player/Player.cpp:691]: Yes but it has do be re-done after equipping, as the equipped items at least for DK change the stats.
I thought about removing the first call that you linked but was unsure about side effects, so I just added a second call at the end to make sure.

Comment (Kitzunu) [reply src/server/game/Entities/Player/Player.cpp:691]: Remove the first in that case

Comment (Kitzunu) [reply src/server/game/Entities/Player/Player.cpp:691]: Doubt it can have any side effects

**Comment by Kitzunu** [comment]
Could you just check that Stam looks okay for DK and maybe Warrior after this change?

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**